Added icon readme and js doc for icon (#479)
etn-ccis/blui-react-native-component-library: docs/Icons.md

Brightlayer UI components will pass these values to you for use in your component to achieve the correct appearance. It is your responsibility to apply them.

> ## Wrapped Icon

Reviewer: You have made same changes here https://github.com/etn-ccis/blui-react-native-component-library/pull/483/files#diff-b1b786098c4fa4ff2f6eea4aa4816179b09676a5b43c6b2f3c3f6920dc9bfe7bL76.

<ComponentName icon={'🍇'} />
```

### Image Source

The icon prop will also support images in various formats, either from local resources or from the web.

```tsx
<Component icon={{ uri: 'https://raw.githubusercontent.com/etn-ccis/blui-icons/dev/packages/png/png48/account_settings_black500_48dp.png' }} />
<Component icon={require('./path/to/image.png')} />
```

## Icon as a component

> `<Icon>` component is an internal utility component used to render icons inside of other components. It standardizes the implementation of the icon and handles all of the different ways to specify the icon without having to duplicate this logic inside of every component that utilizes icons.

Reviewer 2: ```suggestion
`<Icon>` component is an internal utility component used to render icons inside other components. It standardizes the implementation of the icon and handles all of the different ways to specify the icon without having to duplicate this logic inside every component that utilizes icons.
```

> <img width="100%" style="max-width: 300px; display: block;" alt="Icon" src="./images/icon.png">

Reviewer 2: this image is huge

### Usage

```tsx
import { Icon } from '@brightlayer-ui/react-native-components';

<Icon source={{ family: 'brightlayer-ui', name: 'device' }} />;
```

### API

<div style="overflow: auto;">

| Prop Name | Description                        | Type                                   | Required | Default                   |
| --------- | ---------------------------------- | -------------------------------------- | -------- | ------------------------- |

> | source    | A component to render for the icon | `IconSource`                           | yes      |                           |

Reviewer 2: Is "IconSource" a type the same as "Icon Object"'s API? Or are you expecting your user to pass a component, as suggested by the readme?

Author: Yes IconSource type is same as icon object

Reviewer 2: You are still saying "A component to render for the icon" here, which suggests to me that you are expecting adopters to pass something that looks like `<IconComponent ... />` for the `source` prop

```tsx
<Component icon={{ uri: 'https://raw.githubusercontent.com/etn-ccis/blui-icons/dev/packages/png/png48/account_settings_black500_48dp.png' }} />
<Component icon={require('./path/to/image.png')} />
```

## Icon as a component

`<Icon>` component is an internal utility component used to render icons inside other components. It standardizes the implementation of the icon and handles all of the different ways to specify the icon without having to duplicate this logic inside every component that utilizes icons.

<img width="100%" style="max-width: 100px; display: block;" alt="Icon" src="./images/icon.png">

### Usage

```tsx
import { Icon } from '@brightlayer-ui/react-native-components';

<Icon source={{ family: 'brightlayer-ui', name: 'device' }} />;
```

### API

<div style="overflow: auto;">

| Prop Name | Description                                                     | Type                                   | Required | Default                   |
| --------- | --------------------------------------------------------------- | -------------------------------------- | -------- | ------------------------- |

> | source    | An object has the image source or an icon family or JSX.Element | [`IconSource`](#icon-object)           | yes      |                           |

Reviewer 2: you will want to 
* not write this a sentence, but a description of `source`
* make it succinct 

```suggestion
| source    | An object to specify the source of the icon | [`IconSource`](#icon-object)           | yes      |                           |
```
